coremltools/converters/keras/_keras2_converter.py

```diff
@@ -79,12 +79,24 @@ def _is_merge_layer(layer):
                 return True
     return False
 
+def _is_activation_layer(layer):
+    return (isinstance(layer, _keras.layers.core.Activation) or \
+            isinstance(layer, _keras.layers.advanced_activations.LeakyReLU) or \
+            isinstance(layer, _keras.layers.advanced_activations.PReLU) or \
+            isinstance(layer, _keras.layers.advanced_activations.ELU) or \
+            isinstance(layer, _keras.layers.advanced_activations.ThresholdedReLU))
+
 def _check_unsupported_layers(model, add_custom_layers = False):
+    # When add_custom_layers = True, we just convert all layers not present in
+    # registry as custom layer placeholders
+    if add_custom_layers:
+        return
     for i, layer in enumerate(model.layers):
-        if isinstance(layer, _keras.models.Sequential) or isinstance(layer, _keras.models.Model):
+        if isinstance(layer, _keras.models.Sequential) or isinstance(layer,
+                _keras.models.Model):
             _check_unsupported_layers(layer)
         else:
-            if type(layer) not in _KERAS_LAYER_REGISTRY and not add_custom_layers:
```

Comment:
Won't we still need to check this to avoid failing for unknown layers when we want to use a custom layer?

Reply:
I changed that if `add_custom_layers == True`, all unknown layers would be regarded as custom layers, `_check_unsupported_layers` will immediately return.